bitnami/apisix/templates/_helpers.tpl

```diff
@@ -0,0 +1,693 @@
+            sleep "$sleep_time"
+        done
+        return $return_value
+      }
+
+      etcd_hosts=(
+      {{- if .Values.etcd.enabled  }}
+        "{{ ternary "https" "http" $.Values.etcd.auth.client.secureTransport }}://{{ printf "%s:%v" (include "apisix.etcd.fullname" $ ) ( include "apisix.etcd.port" $ ) }}"
+      {{- else }}
+      {{- range $node :=.Values.externalEtcd.servers }}
+        "{{ ternary "https" "http" $.Values.externalEtcd.secureTransport }}://{{ printf "%s:%v" $node (include "apisix.etcd.port" $) }}"
+      {{- end }}
+      {{- end }}
+      )
+
+      check_etcd() {
+          local -r etcd_host="${1:-?missing etcd}"
+          if curl --max-time 5 "${etcd_host}/version" | grep etcdcluster; then
+             return 0
+          else
+             return 1
+          fi
+      }
+
+      for host in "${etcd_hosts[@]}"; do
+          echo "Checking connection to $host"
+          if retry_while "check_etcd $host"; then
+              echo "Connected to $host"
+          else
+              echo "Error connecting to $host"
+              exit 1
+          fi
+      done
+
+      echo "Connection success"
+      exit 0
+{{- end -}}
+
+{{/*
+Init container definition for waiting for the database to be ready
+*/}}
+{{- define "apisix.waitForControlPlaneInitContainer" -}}
+- name: wait-for-control-plane
+  image: {{ template "apisix.wait-container.image" . }}
+  imagePullPolicy: {{ .Values.waitContainer.image.pullPolicy }}
+  {{- if .Values.waitContainer.containerSecurityContext.enabled }}
+  securityContext: {{- omit .Values.waitContainer.containerSecurityContext "enabled" | toYaml | nindent 4 }}
+  {{- end }}
+  command:
+    - bash
+    - -ec
+    - |
+      #!/bin/bash
+      retry_while() {
+        local -r cmd="${1:?cmd is missing}"
+        local -r retries="${2:-12}"
+        local -r sleep_time="${3:-5}"
+        local return_value=1
+
+        read -r -a command <<< "$cmd"
+        for ((i = 1 ; i <= retries ; i+=1 )); do
+            "${command[@]}" && return_value=0 && break
+            sleep "$sleep_time"
+        done
+        return $return_value
+      }
+
+      host="{{ ternary "https" "http" .Values.controlPlane.tls.enabled }}://{{ include "apisix.control-plane.fullname" . }}:{{ .Values.controlPlane.service.ports.adminAPI }}"
+
+      check_control_plane() {
+          if curl --max-time 5 -k -I "$host"; then
+             return 0
+          else
+             return 1
+          fi
+      }
+
+      echo "Checking connection to $host"
+      if retry_while "check_control_plane"; then
+          echo "Connected to $host"
+      else
+          echo "Error connecting to $host"
+          exit 1
+      fi
+
+      echo "Connection success"
+      exit 0
+{{- end -}}
+
+{{- define "apisix.etcd.authEnabled" }}
+{{- if .Values.etcd.enabled -}}
+    {{- if .Values.etcd.auth.rbac.create -}}
+        {{- true -}}
+    {{- end -}}
+{{- else if .Values.externalEtcd.hosts -}}
```

Comment:
The value `.Values.externalEtcd.hosts` does not exist, maybe you meant `externalEtcd.servers`?
```suggestion
{{- else if .Values.externalEtcd.servers -}}
```